### PR TITLE
Test Passers (address previous PR comments)

### PR DESCRIPTION
**Pull request by ak-2485**
Addressed previous PR comments

### CODE REVIEW

File: angular_hit.m [hunk 1]
```diff
@@ -35,57 +36,85 @@
...
 end
 
+
 % Solve the systems Az=b to check for intersection.
+tol = 10^-12;
 Amin = [xmin, -ray_direction(1); ymin, -ray_direction(2)];
 Amax = [xmax, -ray_direction(1); ymax, -ray_direction(2)];
 b = [ray_origin(1)-circle_center(1), ray_origin(2)-circle_center(2)]';
-if (single(tan(min(interval_theta))) == ray_direction(2)/ray_direction(1))
+%if (single(tan(min(interval_theta))) == ray_direction(2)/ray_direction(1))
```
Comment (ak-2485) [inline angular_hit.m]: remove

File: angular_hit.m [hunk 2]
```diff
@@ -35,57 +36,85 @@
...
+if abs(det(Amin)) < tol
     zmax = Amax \ b;
     zmin = [0 ; 0];
-elseif (single(tan(max(interval_theta))) == ray_direction(2)/ray_direction(1))
+%elseif (single(tan(max(interval_theta))) == ray_direction(2)/ray_direction(1))
```
Comment (ak-2485) [inline angular_hit.m]: remove

File: polarCoordinateTraversal.m [hunk 3]
```diff
@@ -111,75 +111,134 @@
...
 current_voxel_ID_r = 1 + (circle_max_radius - r)/delta_radius;
 if verbose
-    fprintf('RADIAL HIT INITIALIZED.\n');
+    fprintf('RADIAL HIT INITIALIZED.\n')
 end
 
-% Calculate Voxel ID Theta.
-current_voxel_ID_theta = floor(atan2(ray_start_y - circle_center_y, ray_start_x - circle_center_x) * num_angular_sections / (2 * pi));
+% II. Calculate Voxel ID Theta.
+tol = 10^-16;
```
Comment (cgyurgyik) [inline polarCoordinateTraversal.m]: Do you not agree we should have a single ```tol``` variable that we pass into other function if they need it?

Comment (ak-2485) [reply polarCoordinateTraversal.m]: yes, we should do that.

File: polarCoordinateTraversal.m [hunk 4]
```diff
@@ -111,75 +111,134 @@
...
+    pert_y = ray_start_y + ray_direction_y * pert_t;
+    current_voxel_ID_theta = floor(atan2(pert_y - circle_center_y, pert_x - circle_center_x) * num_angular_sections / (2 * pi));
+else
+    current_voxel_ID_theta = floor(atan2(ray_start_y - circle_center_y, ray_start_x - circle_center_x) * num_angular_sections / (2 * pi));
+end
 if current_voxel_ID_theta < 0
     current_voxel_ID_theta = num_angular_sections + current_voxel_ID_theta;
 end
 
-% Pre-calculations for radial_hit. This is necessary to check for intersections with relevant radial neighbors.
```
Comment (cgyurgyik) [inline polarCoordinateTraversal.m]: Did you mean to remove this? This was moved to polarCoordinateTraversal in another PR so we didn't calculate it with each radial_hit call.

File: polarCoordinateTraversal.m [hunk 5]
```diff
@@ -111,75 +111,134 @@
...
+    
     % 1. Calculate tMaxR, tMaxTheta
     [tMaxR, tStepR, previous_transition_flag] = radial_hit(ray_origin, ray_direction, ...
         current_voxel_ID_r, circle_center, circle_max_radius, delta_radius, t, ray_unit_vector, ray_circle_vector, v, previous_transition_flag, verbose);
     [tMaxTheta, tStepTheta] = angular_hit(ray_origin, ray_direction, current_voxel_ID_theta,...
         num_angular_sections, circle_center, t, verbose);
     
     % 2. Compare tMaxTheta, tMaxR
-    if (tMaxTheta < tMaxR)
+    if tMaxTheta < tMaxR || current_voxel_ID_r + tStepR == 0 && t < tMaxTheta && tMaxTheta < min(t_grid,t_end)
```
Comment (cgyurgyik) [inline polarCoordinateTraversal.m]: Can we use parentheses here? I don't remember priority for ```&&``` versus```||```, but from my understanding, this code is saying that if the next radial step exits our circle, we should do an angular step, which, correct me if I'm wrong, is not always the case.

Comment (ak-2485) [reply polarCoordinateTraversal.m]: "if the next radial step exits our circle, we should do an angular step" IF the angular step can still be taken within the grid time.

Comment (ak-2485) [reply polarCoordinateTraversal.m]: Order of ops. should be left to right, so parens won't matter: chain of conjunctions

Comment (ak-2485) [reply polarCoordinateTraversal.m]: "if the next radial step exits our circle, we should do an angular step" IF the angular step can still be taken within the grid time. ---- Before, we were exiting the grid automatically dependent on tStepR without checking if we could still take an angular step.

Comment (cgyurgyik) [reply polarCoordinateTraversal.m]: According to the [documentation](https://www.mathworks.com/help/matlab/matlab_prog/operator-precedence.html), there is a precedence order for MATLAB as well.

And ok this makes sense.

File: polarCoordinateTraversal.m (hunk 4 above)
Comment (ak-2485) [reply polarCoordinateTraversal.m]: I meant to remove the comment, yes. The precalcs occur on lines 106, 115, 117
